src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        try {
+            HibernateUtil.getFactory().inTransaction(session ->
+                    session.persist(movieSession));
+            return movieSession;
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't add movieSession " + movieSession, e);
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try {
+            return HibernateUtil.getFactory().fromSession(session ->
+                    Optional.ofNullable(session.get(MovieSession.class, id)));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movieSession by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try {
+            return HibernateUtil.getFactory()
```

Comment:
one question regarding this query - have you checked its performance? will Hibernate fire a single SQL query to database with joins for movies and cinema halls?

by default `@ManyToOne` is eager, which means that for each session movie and hall will be loaded

have you tried this query with `join fetch` instead of eager fetch type? what will be the result?

always guestion yourself when working with a bit more complex relations :)

Reply:
You mean explicitly set FetchType.Lazy for Movie and CinemaHall? I added join fetch to the query but if you mean to set lazy fetch type for movie and cinemaHall I  will make some changes.